### PR TITLE
fix Draw.add validation and add tests

### LINKED ISSUE #387: add()'s trying to fix geojson issues will only make them more complicated
https://github.com/mapbox/mapbox-gl-draw/blob/master/src/api.js#L27-L34

This should be more precise: let's check that the thing is actually a geometry, by testing for `type` being a valid geometry type. Otherwise we're making errors harder to parse by doing too much magic.

### PR DESCRIPTION
**Pull request by mcwhittemore**
Closes #387

### CODE REVIEW

**Comment by mcwhittemore** [comment]
@davidtheclark for the review

File: src/api.js [hunk 1]
```diff
@@ -55,8 +55,15 @@ module.exports = function(ctx) {
       var ids = featureCollection.features.map(feature => {
         feature.id = feature.id || hat();
 
-        if (ctx.store.get(feature.id) === undefined) {
+        if (feature.geometry === null) {
+          throw new Error('Invalid geometry: null');
+        }
+
+        if (ctx.store.get(feature.id) === undefined || ctx.store.get(feature.id).type !== feature.geometry.type) {
```
Comment (davidtheclark) [inline src/api.js]: > ctx.store.get(feature.id).type !== feature.geometry.type

What is that condition meant to catch?

**Comment by davidtheclark** [comment]
Looks good. I had that one question —— We have a use-case for changing the geometry of an existing feature via `Draw.add()`? Is that what it's doing?

**Comment by mcwhittemore** [comment]
@davidtheclark yep.

**Pull request merged**